rocky/rocky/views/mixins.py

```diff
@@ -55,14 +55,24 @@ class OOIAttributeError(AttributeError):
 
 
 class ObservedAtMixin:
-    def get_observed_at(self) -> datetime:
+    @cached_property
+    def observed_at(self) -> datetime:
         observed_at = self.request.GET.get("observed_at", None)
         if not observed_at:
             return datetime.now(timezone.utc)
 
         try:
             datetime_format = "%Y-%m-%d"
-            return convert_date_to_datetime(datetime.strptime(observed_at, datetime_format))
+            date_time = convert_date_to_datetime(datetime.strptime(observed_at, datetime_format))
+            if date_time.date() > datetime.now(timezone.utc).date():
+                messages.warning(
+                    self.request,
+                    _(
+                        "The selected date is in the future. The items below are all items "
+                        "in the database across all existing dates."
```

Comment:
It is not _everything_ before that date and also not "across all existing dates". The objects that are shown are those are valid on the specified date in the future. As I said objects that are currently valid don't have an end date, so are also valid in the future, but previous versions of objects and objects that have been deleted won't be shown, because they have their valid time ends before the selected date.